doc/source/user_guide/text.rst

```diff
@@ -510,7 +589,7 @@ same result as a ``Series.str.extractall`` with a default index (starts from 0).
 
    pd.Index(["a1a2", "b1", "c1"]).str.extractall(two_groups)
 
-   pd.Series(["a1a2", "b1", "c1"]).str.extractall(two_groups)
+   pd.Series(["a1a2", "b1", "c1"], dtype="string").str.extractall(two_groups)
```

Comment:
are we contemplating allowing ``dtype='str'`` and/or ``dtype=str`` here as well? (e.g. the concept of multiple dtype aliases?)

Reply:
we currently map ``str`` -> ``object``, are we changing this? (maybe just in the future)?

Reply:
> we currently map str -> object, are we changing this? (maybe just in the future)?
 
Maybe just in the future. Changing it now would be too disruptive I think.

I'm not super happy about `str` and `'str'` being different from `'string'`, but I think it's unavoidable. We *could* use `'text'` as the string alias, and TextArray / TextDtype to disambiguate a bit?

Reply:
hmm, yeah maybe let's rename this to TextDtype / 'text' I think would be less confusing